common/txmgr/types/tx_store.go

```diff
@@ -109,3 +102,9 @@ type Receipt[R any, TX_HASH types.Hashable, BLOCK_HASH types.Hashable] struct {
 }
 
 type QueryerFunc = func(tx pg.Queryer) error
+
+type ChainReceipt[TX_HASH types.Hashable] interface {
```

Comment:
ah good question - not quite. the `ChainReceipt` interface is a constraint on the chain-specific `R` parameter in `Receipt`. I will add that to make it clearer

Reply:
Sounds good. What I mean is, in a follow-up PR, can't we add BlockHash() and other methods to ChainReceipt, and then just use that everywhere instead of Receipt[...]
Anyways, not for this PR.